dataprofiler/profilers/profile_builder.py

```diff
@@ -602,37 +604,32 @@ def update_profile(self, data, sample_size=None, min_true_samples=None):
         :type min_true_samples
         :return: None
         """
-        if not sample_size:
-            sample_size = self._samples_per_update
         if not min_true_samples:
             min_true_samples = self._min_true_samples
         if isinstance(data, data_readers.base_data.BaseData):
-            self._profile = self._update_profile_from_chunk(
-                data.data, self._profile, sample_size, min_true_samples, self.options)
+            self._update_profile_from_chunk(
+                data.data, sample_size, min_true_samples, self.options)
             self._update_row_statistics(data.data)
             self.encoding = data.file_encoding
             self.file_type = data.data_type
         elif isinstance(data, pd.DataFrame):
-            self._profile = self._update_profile_from_chunk(
-                data, self._profile, sample_size, min_true_samples, self.options)
+            self._update_profile_from_chunk(
+                data, sample_size, min_true_samples, self.options)
             self._update_row_statistics(data)
             self.file_type = str(data.__class__)
         else:
             raise ValueError(
                 "Data must either be imported using the data_readers or "
                 "pd.DataFrame."
             )
-   
-    @staticmethod
-    def _update_profile_from_chunk(df, profile=None, sample_size=None,
+
+    def _update_profile_from_chunk(self, df, sample_size=None,
```

Comment:
previously `_update_profile_from_chunk` was static, but it doesn't really make sense and means we no longer have to pass profiles and can just use `self._profile` within the function

This allows us to get the sample_size within `_update_profile_from_chunk`

Reply:
I'm no recalling the original intent was because if something error'ed within update, we wouldn't update anything. We can revert to keep this intent if desired.